osaca/semantics/kernel_dg.py

```diff
@@ -365,62 +376,68 @@ def is_read(self, register, instruction_form):
         if instruction_form.semantic_operands is None:
             return is_read
         for src in chain(
-            instruction_form.semantic_operands.source,
-            instruction_form.semantic_operands.src_dst,
+            instruction_form.semantic_operands["source"],
+            instruction_form.semantic_operands["src_dst"],
         ):
-            if "register" in src:
-                is_read = self.parser.is_reg_dependend_of(register, src.register) or is_read
-            if "flag" in src:
-                is_read = self.parser.is_flag_dependend_of(register, src.flag) or is_read
-            if "memory" in src:
-                if src.memory.base is not None:
-                    is_read = self.parser.is_reg_dependend_of(register, src.memory.base) or is_read
-                if src.memory.index is not None:
-                    is_read = (
-                        self.parser.is_reg_dependend_of(register, src.memory.index) or is_read
-                    )
+            if isinstance(src, RegisterOperand):
+                is_read = self.parser.is_reg_dependend_of(register, src) or is_read
+            if not isinstance(src, Operand) and (
+                "flag" in src or src["class"] == "flag" if "class" in src else False
```

Comment:
I think flags should be also operands. So we should have a class `FlagOperand` that derives from `Operand`

Reply:
https://github.com/RRZE-HPC/OSACA/commit/fa95293cb013ca502f231bc60991f5c419b650f4